expo/expo-cli
# fix --config option when evaluating config

fix https://github.com/expo/expo-cli/issues/1542 

When a custom config path is used, the expo object isn't reduced out of the config. 

# Test Plan

- Add test to ensure custom config paths return an object with the "expo" field reduced when reading JSON objects.
- Add test to ensure the "expo" object is only reduced once.
